### ISSUE TITLE
[UWP] FrameAdapter cannot remove event handler registered to frame 

### ISSUE THREAD

**Issue by grapehunter** [Caliburn-Micro/Caliburn.Micro#356]
1,Register instance with frame

``` csharp
var frameAdapter = new FrameAdapter(frame);
IocContainer.RegisterInstance(typeof(INavigationService), frame.Name, frameAdapter);
```

2,After page which the frame in is never used, try to unregister handle

``` csharp
IocContainer.UnregisterHandler(typeof(INavigationService), "Frame Name");
```

3,The instance of frameAdapter will never be released , because 3 event handler below was not be removed.(And it means that the page will never be deconstructed too)

``` csharp
this.frame.Navigating += OnNavigating;
this.frame.Navigated += OnNavigated;
var navigationManager = SystemNavigationManager.GetForCurrentView();
 navigationManager.BackRequested += OnBackRequested;
```

**Comment by grapehunter** [comment]
I think the three event handlers should be removed when frame was unloading.

**Comment by nigel-sampson** [comment]
Thinking about this I suspect the approach will be to make FrameAdapter disposable to detach from the events.

I've found that depending on Loaded / Unloaded events to fraught with problems around timing and so on.

Thoughts?

**Comment by grapehunter** [comment]
Sorry for the late reply.
This issue was very important to me . So I create a copy for **FrameAdapter** to the local , and moditied the code to remove event handler when frame unloading.

**Comment by grapehunter** [comment]
![image](https://user-images.githubusercontent.com/3354458/26872017-b07154f8-4ba7-11e7-94d7-7e0fca3ddacb.png)

There was two frames in MainView . After user logging out or changing theme/language ,I navigate to MainBridge or LoginView . But because of the memory leak, the MainView will never be released.

**Comment by nigel-sampson** [comment]
No worries, sounds like you're approach is working which is good.

There definitely does need to be some work done here, I'm not 100% sold on the events being registered / deregistered on `Loaded` and `Unloaded` due to it's vulnerability to timing issues (navigation service created after the `Frame` is loaded for instance).

**Comment by grapehunter** [comment]
Very glad to see that this feature has been added into milestone v4.
Now I'm going to use my own modified `FrameAdapter ` instead.

**Comment by KasperSK** [comment]
Should this be closed since it was fixed in pull request #728?

**Comment by grapehunter** [comment]
Nice job, Thank you very much
